Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/MenuViewController.swift

```diff
@@ -1,405 +1,287 @@
+                self.panModalSetNeedsLayoutUpdate()
             }
         }
-        
-        var icon: UIImage {
-            switch self {
-            case .vpn: return #imageLiteral(resourceName: "vpn_menu_icon").template
-            case .bookmarks: return #imageLiteral(resourceName: "menu_bookmarks").template
-            case .history: return #imageLiteral(resourceName: "menu-history").template
-            case .settings: return #imageLiteral(resourceName: "menu-settings").template
-            case .add: return #imageLiteral(resourceName: "menu-add-bookmark").template
-            case .share: return #imageLiteral(resourceName: "nav-share").template
-            case .downloads: return #imageLiteral(resourceName: "menu-downloads").template
+        present(container, animated: true) {
+            self.panModalSetNeedsLayoutUpdate()
+        }
+        if expandToLongForm {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.15) {
+                self.panModalTransition(to: .longForm)
             }
         }
     }
     
-    private let bvc: BrowserViewController
-    private let tab: Tab?
-    
-    /// Keeping reference to it to monitor vpn status and set correct `isEnabled` status.
-    private var vpnMenuCell: MenuCell?
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationBar.isTranslucent = false
+    }
     
-    private lazy var visibleButtons: [MenuButtons] = {
-        let allButtons = MenuButtons.allCases
-        
-        // Don't show url buttons if there is no url to pick(like on home screen)
-        var allWithoutUrlButtons = allButtons
-        allWithoutUrlButtons.removeAll { $0 == .add || $0 == .share }
-        
-        guard let url = tab?.url, (!url.isLocal || url.isReaderModeURL) else {
-            return allWithoutUrlButtons
+    override func viewSafeAreaInsetsDidChange() {
+        super.viewSafeAreaInsetsDidChange()
+        
+        // Bug with pan modal + hidden nav bar causes safe area insets to zero out
+        if view.safeAreaInsets == .zero, isPanModalPresented,
+           var insets = view.window?.safeAreaInsets {
+            // When that happens we re-set them via additionalSafeAreaInsets to the windows safe
+            // area insets. Since the pan modal appears over the entire screen we can safely use
+            // the windows safe area. Top will stay 0 since we are using non-translucent nav bar
+            // and the top never reachs the safe area (handled by pan modal)
+            insets.top = 0
+            additionalSafeAreaInsets = insets
         }
-        return allButtons
-    }()
-    
-    // MARK: - Init
-    
-    init(bvc: BrowserViewController, tab: Tab?) {
-        self.bvc = bvc
-        self.tab = tab
-
-        super.init(nibName: nil, bundle: nil)
     }
     
-    @available(*, unavailable)
-    required init?(coder aDecoder: NSCoder) {
-        fatalError()
+    override var preferredStatusBarStyle: UIStatusBarStyle {
```

Comment:
Just curious do we need need to set Status bar style for this controller ?

Reply:
It is to keep the status bar white when the user navigates to other VC's inside the menu